src/Reveal.Sdk.Dom.Tests/Visualizations/TimeSeriesVisualizationFixture.cs

```diff
@@ -0,0 +1,402 @@
+        Assert.Null(visualization.Linker);
+        Assert.Equal(0, visualization.RowSpan);
+        Assert.NotNull(visualization.Settings);
+        Assert.IsType<TimeSeriesVisualizationSettings>(visualization.Settings);
+        Assert.Null(visualization.Title);
+        Assert.NotNull(visualization.Values);
+        Assert.Empty(visualization.Values);
+        Assert.NotNull(visualization.VisualizationDataSpec);
+        Assert.IsType<TimeSeriesVisualizationDataSpec>(visualization.VisualizationDataSpec);
+    }
+
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void Constructor_InitializesTimeSeriesChartVisualizationWithDataSourceItem_WhenDataSourceItemIsProvided(
+        bool hasTabularData)
+    {
+        // Arrange
+        var dataSourceItem = new DataSourceItem { HasTabularData = hasTabularData };
+
+        // Act
+        var timeSeriesVisualization = new TimeSeriesVisualization(dataSourceItem);
+
+        // Assert
+        Assert.NotNull(timeSeriesVisualization);
+        Assert.Equal(ChartType.TimeSeries, timeSeriesVisualization.ChartType);
+        Assert.Equal(dataSourceItem, timeSeriesVisualization.DataDefinition.DataSourceItem);
+        Assert.Null(timeSeriesVisualization.Title);
+        Assert.Equal(hasTabularData, timeSeriesVisualization.DataDefinition.DataSourceItem.HasTabularData);
+    }
+
+    [Theory]
+    [InlineData("Test Title", null, null)]
+    [InlineData(null, null, null)]
+    [InlineData("Test Title with Data Source", true, true)]
+    [InlineData("Test Title without Tabular Data", false, false)]
+    public void Constructor_SetsTitleAndDataSourceItem_WhenArgumentsAreProvided(string title, bool? hasTabularData,
+        bool? expectedHasTabularData)
+    {
+        // Arrange
+        var dataSourceItem = hasTabularData.HasValue
+            ? new DataSourceItem { HasTabularData = hasTabularData.Value }
+            : null;
+        // Act
+        var timeSeriesVisualization = new TimeSeriesVisualization(title, dataSourceItem);
+
+        // Assert
+        Assert.Equal(title, timeSeriesVisualization.Title);
+        Assert.Equal(ChartType.TimeSeries, timeSeriesVisualization.ChartType);
+
+        if (dataSourceItem == null)
+        {
+            Assert.Null(timeSeriesVisualization.DataDefinition);
+        }
+        else
+        {
+            Assert.NotNull(timeSeriesVisualization.DataDefinition);
+            Assert.Equal(expectedHasTabularData, timeSeriesVisualization.DataDefinition.DataSourceItem.HasTabularData);
+        }
+    }
+
+    [Fact]
+    public void Date_CanBeSetAndRetrieved_WhenInitialized()
+    {
+        // Arrange
+        var dateColumn = new DimensionColumn { DataField = new MockDimensionDataField("DateField") };
+        var visualization = new TimeSeriesVisualization();
+
+        // Act
+        visualization.Date = dateColumn;
+
+        // Assert
+        Assert.Equal(dateColumn, visualization.Date);
```

Comment:
I think we should check the case that the VisualizationDataSpec.Rows is blank, i.e, check for the case that the Date getter reuturns null